### PR TITLE
Add Docker instructions to README

### PR DESCRIPTION
**Pull request by VHellendoorn**
Provides the instructions for using the `Dockerfile` to run the code, as discussed in #419. Feedback welcome; hope this is helpful.

### CODE REVIEW

**Comment by VHellendoorn** [comment]
Absolutely! I'm currently training a models at around the 1B parameter mark (e.g., from `large.yml`) on roughly 150GB of data on a single node with 8 RTX8000 GPUs. At that scale, I'm seeing about 45 TFLOPS per GPU, which translates into ~18 sequences of 2,048 tokens per second on the full node. With better memory utilization, I can push the latter to about 25. That about echoes the realistic maximum these cards output and near-ideal node-level parallelism (about 7x the per-GPU performance); utilization is at a steady 100%.

I am planning to venture into the few-billion parameter territory after this, but realistically I need more than one server to train at that level. In my own benchmarking, I can get a throughput of about 7 sequences per second on the 2-7B model, so (with gradient-accumulation to accomodate larger batch sizes) that means training about 3x slower (makes sense; it's 3x the parameters). I'm already not getting through much more than 5GB of data per day, so training a larger model once on 150GB would take many months.

In terms of configuration, I tried a number of sweeps on the `large` configuration to calibrate. The results is that I am not currently using the model or pipeline parallelism -- the former slowed training down dramatically (maybe because no NVLink) and the latter was about even, though worse for larger batch sizes (probably because this is all intra-node). I do always enable the two flags for softmax and gelu fusion; they tend to add neglible setup time and boosted performance by a pretty significant amount in a few ad-hoc experiments. The stage-1 ZeRO optimizations seem to help as well; higher stages don't really seem to pay off, but maybe they will yet. And 1-bit Adam had no discernible impact whatsoever, which again makes some sense for single-node training. Oh and sparse training did speed things up a good bit, but led to errors when using in text-generation mode, some of which I could fix with padding and others that seemed more fundamental. I'm deferring on that one for a while.

I think that captures all my experiments. Hope this helps!
